From 05223364f628fe87f1e07ace80a67c89788297f5 Mon Sep 17 00:00:00 2001
From: Ganapatrao Kulkarni <gkulkarni@caviumnetworks.com>
Date: Wed, 11 May 2016 19:58:02 -0400
Subject: [PATCH 24/39] arm64: prefetch: enable ARM64_HAS_NO_HW_PREFETCH for
 thunderx 81xx.

163373d arm64: prefetch: add alternative pattern for CPUs without a prefetcher
is applicable for thunderx-81xx as well.

Signed-off-by: Ganapatrao Kulkarni <gkulkarni@caviumnetworks.com>
---
 arch/arm64/include/asm/cputype.h | 2 ++
 arch/arm64/kernel/cpufeature.c   | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/include/asm/cputype.h b/arch/arm64/include/asm/cputype.h
index 7540284..e3341d0 100644
--- a/arch/arm64/include/asm/cputype.h
+++ b/arch/arm64/include/asm/cputype.h
@@ -85,10 +85,12 @@
 #define APM_CPU_PART_POTENZA		0x000
 
 #define CAVIUM_CPU_PART_THUNDERX	0x0A1
+#define CAVIUM_CPU_PART_THUNDERX_81XX	0x0A2
 
 #define MIDR_CORTEX_A53 MIDR_CPU_MODEL(ARM_CPU_IMP_ARM, ARM_CPU_PART_CORTEX_A53)
 #define MIDR_CORTEX_A57 MIDR_CPU_MODEL(ARM_CPU_IMP_ARM, ARM_CPU_PART_CORTEX_A57)
 #define MIDR_THUNDERX	MIDR_CPU_MODEL(ARM_CPU_IMP_CAVIUM, CAVIUM_CPU_PART_THUNDERX)
+#define MIDR_THUNDERX_81XX	MIDR_CPU_MODEL(ARM_CPU_IMP_CAVIUM, CAVIUM_CPU_PART_THUNDERX_81XX)
 
 #ifndef __ASSEMBLY__
 
diff --git a/arch/arm64/kernel/cpufeature.c b/arch/arm64/kernel/cpufeature.c
index aaf16c2..a62c871 100644
--- a/arch/arm64/kernel/cpufeature.c
+++ b/arch/arm64/kernel/cpufeature.c
@@ -630,7 +630,8 @@ static bool has_no_hw_prefetch(const struct arm64_cpu_capabilities *entry)
 	rv_min = 0;
 	rv_max = (1 << MIDR_VARIANT_SHIFT) | MIDR_REVISION_MASK;
 
-	return MIDR_IS_CPU_MODEL_RANGE(midr, MIDR_THUNDERX, rv_min, rv_max);
+	return MIDR_IS_CPU_MODEL_RANGE(midr, MIDR_THUNDERX, rv_min, rv_max) ||
+		MIDR_IS_CPU_MODEL_RANGE(midr, MIDR_THUNDERX_81XX, rv_min, rv_max);
 }
 
 static const struct arm64_cpu_capabilities arm64_features[] = {
-- 
1.9.1

